aurora/src/main/java/com/market/banica/aurora/mapper/RequestMapper.java

```diff
@@ -71,18 +76,55 @@ public RequestMapper(StubManager stubManager, ChannelManager channelManager) {
             return this.renderOrderbookMapping(incomingRequest, channelByKey.get());
         } else if (destinationOfMessage.contains(AURORA)) {
             return renderAuroraMapping(incomingRequest, channelByKey.get());
-        } else if (destinationOfMessage.contains(MARKET)){
-            throw new ServiceNotFoundException("Unimplemented publisher!");
+        } else if (destinationOfMessage.contains(MARKET)) {
+            return renderMarketMapping(incomingRequest, channelByKey.get());
         }
 
         throw new ServiceNotFoundException(BAD_PUBLISHER_REQUEST + ". Requested publisher is: " + destinationOfMessage);
     }
 
+    private Aurora.AuroraResponse renderMarketMapping(Aurora.AuroraRequest incomingRequest, ManagedChannel channelByKey) {
+        LOGGER.debug("Mapping messages for market.");
+        String[] topicSplit = incomingRequest.getTopic().split(SPLIT_SLASH_REGEX);
+
+        MarketServiceGrpc.MarketServiceBlockingStub marketStub = stubManager.getMarketBlockingStub(channelByKey);
+
+        String itemName = topicSplit[2];
+        double itemPrice = Double.parseDouble(topicSplit[3]);
+        long itemQuantity = Long.parseLong(topicSplit[4]);
+        String marketName = topicSplit[5];
+
+        ProductBuySellRequest.Builder request = ProductBuySellRequest.newBuilder()
+                .setItemName(itemName)
+                .setItemPrice(itemPrice)
+                .setItemQuantity(itemQuantity)
+                .setMarketName(marketName);
+
+        String requestPrefix = topicSplit[1];
+
+        if (topicSplit.length == 6 && requestPrefix.equals(AVAILABILITY_ACTION)) {
```

Comment:
Do we have a case when topicSplit.length == 6 and requestPrefix != AVAILABILITY_ACTION ? if so this will cause arrayindex out of bound exception.

Reply:
Fixed